autoload/ale/fixers/ormolu.vim

```diff
@@ -1,12 +1,20 @@
 call ale#Set('haskell_ormolu_executable', 'ormolu')
 call ale#Set('haskell_ormolu_options', '')
 
-function! ale#fixers#ormolu#Fix(buffer) abort
+function! ale#fixers#ormolu#GetExecutable(buffer) abort
     let l:executable = ale#Var(a:buffer, 'haskell_ormolu_executable')
+
+    return ale#handlers#haskell_stack#EscapeExecutable(l:executable, 'ormolu')
+endfunction
+
+function! ale#fixers#ormolu#Fix(buffer) abort
+    let l:executable = ale#fixers#ormolu#GetExecutable(a:buffer)
     let l:options = ale#Var(a:buffer, 'haskell_ormolu_options')
 
     return {
-    \   'command': ale#Escape(l:executable)
-    \       . (empty(l:options) ? '' : ' ' . l:options),
+    \   'command': l:executable
+    \       . (empty(l:options) ? '' : ' ' . l:options)
+    \       . ' --stdin-input-file '
+    \       . ale#Escape(@%),
```

Comment:
`@%` may not be the current file at the time the linter runs, as this code could be executed from a different buffer. Instead get the filename using the `a:buffer` variable. ALE has a shorthand for this where you can use `%s` in the command string. See `:help ale-command-format-strings`.

Have a look at changelogs and make sure the `--stdin-input-file` argument is available in old enough `ormolu` versions. If it's not available in older versions, you can make this fixer function still compatible with older versions with the `ale#semver` functions and a version check. You can grep the codebase for other fixers that run version checks.

Reply:
Sorry for being slow to respond but thanks for the feedback!

> @% may not be the current file at the time the linter runs, as this code could be executed from a different buffer. Instead get the filename using the a:buffer variable. ALE has a shorthand for this where you can use %s in the command string. See :help ale-command-format-strings.

👍

> Have a look at changelogs and make sure the --stdin-input-file argument is available in old enough ormolu versions. If it's not available in older versions, you can make this fixer function still compatible with older versions with the ale#semver functions and a version check. You can grep the codebase for other fixers that run version checks.

`ormolu` is a Haskell program which uses [Haskell PVP](https://pvp.haskell.org/) which AFAIU means the `ale#semver` functions won't work. I couldn't find usage of PVP version checks in the codebase, but I guess one option would be to create a similar `ale#pvp` module. Does that sound reasonable or do you know of a better way to solve this?